src/app/pages/products-list/products-list.component.ts

```diff
@@ -14,11 +15,22 @@ export class ProductsListComponent implements OnInit {
 
     constructor(
         private readonly productsStoreService: ProductsStoreService,
+        private readonly activatedRoute: ActivatedRoute,
         private readonly router: Router,
     ) {}
 
     ngOnInit(): void {
-        this.productsStoreService.loadProducts();
+        this.activatedRoute.paramMap.pipe(
+            map(paramMap=>{
+                return paramMap.get("id")
+            }),
+        ).subscribe(subCategoryId=>{
+            if(subCategoryId){
+                this.productsStoreService.loadProducts(subCategoryId)
+                return
+            }
+            this.productsStoreService.loadProducts()
+        })
```

Comment:
Почему не просто передаешь `subCategoryId` в метод? Он же умеет работать с `null`🤔
```suggestion
            this.productsStoreService.loadProducts(subCategoryId)
        })
```